### PR TITLE
Refactor stream handling in LiveViewTest

### PR DESCRIPTION
**Pull request by SteffenDE**
This PR refactors the handling of streams in LiveViewTest to make it easier to understand. I mirrored the handling of streams in the JavaScript client, minus the morphdom related stuff.

It also adds a clause that *should*  handle the case of nested LiveViews where the LV parent updates. Currently, streams inside the child LV would be cleared. This *may* be a solution for that, although I'm not sure.

### CODE REVIEW

File: lib/phoenix_live_view/test/dom.ex [hunk 1]
```diff
@@ -426,80 +399,6 @@ defmodule Phoenix.LiveViewTest.DOM do
...
-                  child -> set_attr(child, "data-phx-stream", ref)
-                end
-
-              parent_id = parent_id(html_tree, id)
-
-              cond do
-                !child ->
-                  acc
-
-                # skip added children that aren't ours if they are not being appended
```
Comment (SteffenDE) [inline lib/phoenix_live_view/test/dom.ex:472]: To be discussed: how are we handling non-stream items inside a stream container? I don't think this is something that we officially support, right? I that case we could also check for non stream elements and warn users.

Comment (josevalim) [reply lib/phoenix_live_view/test/dom.ex:472]: Should we raise ? so we force people to fix both for the benefit of both JS and Elixir clients?

Comment (SteffenDE) [reply lib/phoenix_live_view/test/dom.ex:472]: Something like this maybe: https://github.com/phoenixframework/phoenix_live_view/pull/3078/commits/a4e0b8024fbd6db96fd42ff28d43020717dbd1c0

image

**Comment by SteffenDE** [comment]
Found an issue either in the code or in our app at work. Will check tomorrow.

**Comment by SteffenDE** [comment]
The issue is again with nested LiveViews... will investigate more

File: lib/phoenix_live_view/test/dom.ex (hunk 1 above)
Comment (chrismccord) [reply lib/phoenix_live_view/test/dom.ex:472]: Confirm we do not support non stream items in a stream container so this error looks good!

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Comment by dspigarelli** [comment]
Too bad @clarkware has been [promoting this](https://twitter.com/clarkware/status/1705263936491356325) as a clever technique for handling empty streams.

**Comment by chrismccord** [comment]
@dspigarelli we're going to make that technique work :)

**Comment by Munksgaard** [comment]
> @dspigarelli we're going to make that technique work :)

Any news on this? I don't see an tracking issue, should we create one?

**Comment by Munksgaard** [comment]
@chrismccord Any news here?

**Comment by SteffenDE** [comment]
Sorry @Munksgaard; I created #3260 to track this.

**Comment by Munksgaard** [comment]
No worries, and thank you! :heart:

**Pull request merged**